### DISCUSSION TITLE
Resolving Cluster IP changes in AWS

### DISCUSSION THREAD

**Discussion by clintz** [redis/rueidis#185] [General]
Hi - 
Great project. Indeed much faster than go-redis. 

I am curious if the client resolves potentially changing IPs when using in AWS. MemoryDB for example exposes a hostname endpoint that resolves to all the node in the cluster. From the docs it appears I would have to resolve the IPs and plug them into when doing NewClient. Is there any magic going on under the hood where hostnames are resolved to IPs and changes going forward are picked up? 

Thanks!

**Comment by rueian** [comment]
Hi @clintz,

You don't need to resolve IPs manually. Rueidis will use your DNS to resolve hostname and follow [redis cluster redirection spec](https://redis.io/docs/reference/cluster-spec/#redirection-and-resharding) to pick up new IPs. So, you can just plug the ClusterEndpoint of your MemoryDB to rueidis.NewClient.
